cdap-gateway/src/test/java/co/cask/cdap/gateway/handlers/metrics/MetricsHandlerTestRun.java

```diff
@@ -381,8 +383,133 @@ public void testQueryMetrics() throws Exception {
   }
 
   @Test
-  public void testQueryMetricsWithTags() throws Exception {
+  public void testAggregateQueryBatch() throws Exception {
+
+    Map<String, List<QueryRequest>> batchQueryInput = Maps.newHashMap();
```

Comment:
on the api: we take in map of query_id->query and return back query_id->query_result